File: API/Helpers/UserParams.cs
namespace API;

public class UserParams : PaginationParams
{
    public string CurrentUsername { get; set; }
    public string Gender { get; set; }
    public int MinAge { get; set; } = 0;
    public int MaxAge { get; set; } = 18;
    public string OrderBy { get; set; } = "lastActive";
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 0;
    maxAge = 18;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lastActive';

    constructor(user: User) {
        // this.gender  = user.gender === 'female' ? 'male' : 'female'
        this.gender = 'all';
    }
}
